packages/boilerplate/theme/components/Checkout/UserBillingAddresses.vue

```diff
@@ -0,0 +1,95 @@
+<template>
+  <div>
+    <SfAddressPicker
+      :value="currentAddressId"
+      @input="setCurrentAddress($event)"
+      class="billing__addresses"
+    >
+      <SfAddress
+        v-for="billingAddress in billingAddresses"
+        :key="userBillingGetters.getId(billingAddress)"
+        :name="String(userBillingGetters.getId(billingAddress))"
+      >
+        <span
+          >{{ userBillingGetters.getFirstName(billingAddress) }} {{ userBillingGetters.getLastName(billingAddress) }}</span
+        >
+        <span
+          >{{ userBillingGetters.getStreetName(billingAddress) }}
+          {{ userBillingGetters.getApartmentNumber(billingAddress) }}</span
+        >
+        <span>{{ userBillingGetters.getPostCode(billingAddress) }}</span>
+        <span
+          >{{ userBillingGetters.getCity(billingAddress)
+          }}{{ userBillingGetters.getProvince(billingAddress) ? `, ${userBillingGetters.getProvince(billingAddress)}` : '' }}</span
+        >
+        <span>{{ userBillingGetters.getCountry(billingAddress)}}</span>
+        <span>{{ userBillingGetters.getPhone(billingAddress) }}</span>
+      </SfAddress>
+    </SfAddressPicker>
+    <SfCheckbox
+      data-cy="billing-details-checkbox_isDefault"
+      v-model="localSetAsDefault"
+      name="setAsDefault"
+      label="Use this address as my default one."
+      class="billing-address-setAsDefault"
+    />
+  </div>
+</template>
+
+<script>
+import {
+  SfCheckbox
+} from '@storefront-ui/vue';
+import SfAddressPicker from '~/components/temp/SfAddressPicker';
+import { ref, watch } from '@vue/composition-api';
+import { userBillingGetters } from '@vue-storefront/boilerplate';
+
+export default {
+  name: 'UserBillingAddresses',
+  props: {
+    currentAddressId: {
+      type: Number,
+      required: true
+    },
+    setAsDefault: {
+      type: Boolean,
+      required: true
+    },
+    billingAddresses: {
+      type: Array,
+      required: true
+    }
+  },
+  components: {
+    SfCheckbox,
+    SfAddressPicker
+  },
+  watch: {
+    setAsDefault (setAsDefault) {
+      this.localSetAsDefault = setAsDefault;
+    }
+  },
+  setup ({ setAsDefault }, { emit }) {
+    const setCurrentAddress = $event => emit('setCurrentAddress', $event);
+    const localSetAsDefault = ref(setAsDefault);
+    watch(localSetAsDefault, () => emit('changeSetAsDefault', localSetAsDefault.value));
```

Comment:
you don't need both watchers - listen on @input or @change in the SfCheckbox

Reply:
when it depends on the prop it's quite weird when you listen on that in the child and then update again in the parent - I think you can avoid such a watcher

Reply:
Indeed